pyodide_build/out_of_tree/build.py

```diff
@@ -28,6 +31,19 @@ def run(
     env = os.environ.copy()
     env.update(build_env.get_build_environment_vars(get_pyodide_root()))
 
+    # create a persistent build dir in the source dir
+    build_dir = srcdir / ".pyodide_build"
+    build_dir.mkdir(exist_ok=True)
+    # don't track the build dir (helps if building in a git repo)
+    gitignore_path = build_dir / ".gitignore"
+    if not gitignore_path.exists():
+        with gitignore_path.open("w") as f:
+            f.write(
+                dedent("""\
+                    # Created by pyodide-build
+                    *""")
+            )
```

Comment:
Could you extract this out to a function, such as `prepare_build_dir(build_dir)`?

Reply:
Yes, this makes sense – thanks. I extracted it out to `_prepare_build_dir` and also added a VCS-ignore file for Mercurial in de93d2b62c6119ea245dbeeb674d0b0301f20bca and 6ef79be4e11702e9139af8001b989859548ea2d8 respectively.